lib/mix/phoenix/context.ex

```diff
@@ -25,12 +25,12 @@ defmodule Mix.Phoenix.Context do
     ctx_app   = opts[:context_app] || Mix.Phoenix.context_app()
     base      = Module.concat([Mix.Phoenix.context_base(ctx_app)])
     module    = Module.concat(base, context_name)
-    alias     = module |> Module.split() |> tl() |> Module.concat()
-    basename  = Phoenix.Naming.underscore(context_name)
-    dir       = Mix.Phoenix.context_lib_path(ctx_app, basename)
+    alias     = module |> Module.split() |> List.last() |> Module.concat(nil)
+    basename  = context_name |> String.split(".") |> List.last |> Phoenix.Naming.underscore
+    dir       = Mix.Phoenix.context_lib_path(ctx_app, Phoenix.Naming.underscore(context_name))
```

Comment:
In the case of a context being `Site.Blog`, after underscoring it turns into `site/blog` which gives us the desired path